src/components/charts-generic/histogram/histogram-state.tsx

```diff
@@ -219,9 +229,29 @@ const useHistogramState = ({
     : [{ height: 0, nbOfLines: 1 }];
 
   const getAnnotationInfo = (d: Bin<GenericObservation, number>) => {
+    let xAnchor;
+    let binIndex = 0;
+    let meta = undefined;
+    if (groupedBy && binMeta) {
+      const bandDomain = binMeta.map((b, i) => b.label ?? String(i));
+      const bandScale = scaleBand().domain(bandDomain).range([0, chartWidth]);
```

Comment:
It seem s the band Scale should not be created in the annotation info but before.